packages/vite/src/node/plugins/html.ts

```diff
@@ -45,7 +45,7 @@ interface ScriptAssetsUrl {
   url: string
 }
 
-const htmlProxyRE = /\?html-proxy=?[&inline\-css]*&index=(\d+)\.(js|css)$/
+const htmlProxyRE = /\?html-proxy=?(&inline-css)*&index=(\d+)\.(js|css)$/
```

Comment:
Great eye. I think the previous was indeed incorrect but we need a bit more changes here:

```suggestion
const htmlProxyRE = /\?html-proxy=?(?:&inline-css)?&index=(\d+)\.(js|css)$/
```

Made a a non-capturing group since we index the `\d+` part in-code. And used `?` instead of `*` as we want to match one optional instance of it.

Reply:
Yes, I fix it.